src/dependency/Pyegi/Pyegi/installer.py

```diff
@@ -0,0 +1,169 @@
+    script_path = scriptsPath + script + "/"
+    if exists(script_path + "pyproject.toml"):
+        os.chdir(script_path)
+        os.system("poetry lock")
+        src = script_path + "pyproject.toml"
+        dst = temp_dir + "pyproject.toml"
+        shutil.copyfile(src, dst)
+        os.chdir(temp_dir)
+        os.system("poetry lock")
+        lock_content = toml.load("poetry.lock")
+        packages = lock_content["package"]
+        packages_to_move = []
+        phase1_start_time = dt1.now()
+        for package in packages:
+            name_in_commons = f"{package['name']}-{package['version']}"
+            f = open(commons_dir + "lib_links.json")
+            lib_links = json.load(f)
+            f.close()
+            new_package = True
+            for pkg in lib_links["Packages"]:
+                if pkg["name"] == name_in_commons:
+                    if script not in pkg["scripts"]:
+                        pkg["scripts"].append(script)
+                    new_package = False
+            if new_package:
+                lib_link = {}
+                lib_link["name"] = name_in_commons
+                lib_link["scripts"] = [script]
+                lib_links["Packages"].append(lib_link)
+            json.dump(lib_links, open(commons_dir + "lib_links.json", "w"))
+            isdir = os.path.isdir(commons_dir + name_in_commons)
+            if isdir:
+                filename = f"{commons_dir + name_in_commons}/Lib/site-packages/{name_in_commons}.dist-info/RECORD"
+                paths = []
+                with open(filename, "r") as csvfile:
+                    csvreader = csv.reader(csvfile)
+                    for row in csvreader:
+                        paths.append(row[0])
+                for path in paths:
+                    connection_path = "/Lib/site-packages/"
+                    if path[:3] == "../":
+                        connection_path = "/Lib/"
+                        path = path[3:]
+                        if path[:3] == "../":
+                            connection_path = "/"
+                            path = path[3:]
+                    src = commons_dir + name_in_commons + connection_path + path
+                    dst = f"{temp_dir}.venv{connection_path + path}"
+                    create_dirs(dst)
+                    os.symlink(src, dst)
+                    dst = f"{script_path}.venv{connection_path + path}"
+                    create_dirs(dst)
+                    os.symlink(src, dst)
+            else:
+                packages_to_move.append(name_in_commons)
+        phase1_end_time = dt1.now()
+        print("Total phase 1 time: " + str(phase1_end_time - phase1_start_time))
+
+        os.system("poetry install")
+        phase2_start_time = dt1.now()
+        for name_in_commons in packages_to_move:
+            filename = (
+                f"{temp_dir}.venv/Lib/site-packages/{name_in_commons}.dist-info/RECORD"
+            )
+            if exists(filename):
```

Comment:
maybe the `else` should raise a warning message, which says "package name_in_commons didn't have any RECORD file"

Reply:
until we figure out why this thing happens I suggest we keep it this way

Reply:
that's exactly why we should warn it and don't hide it. if we hide it, we never notice the reason.